x-pack/plugins/ml/public/application/components/annotations/annotation_flyout/index.test.tsx

```diff
@@ -5,58 +5,96 @@
  */
 
 import useObservable from 'react-use/lib/useObservable';
-
 import mockAnnotations from '../annotations_table/__mocks__/mock_annotations.json';
-
 import React from 'react';
-import { mountWithIntl, shallowWithIntl } from '@kbn/test/jest';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
 
 import { Annotation } from '../../../../../common/types/annotations';
-import { annotation$ } from '../../../services/annotations_service';
+import { AnnotationUpdatesService } from '../../../services/annotations_service';
 
 import { AnnotationFlyout } from './index';
+import { MlAnnotationUpdatesContext } from '../../../contexts/ml/use_ml_annotation_updates';
 
-describe('AnnotationFlyout', () => {
-  test('Initialization.', () => {
-    const wrapper = shallowWithIntl(<AnnotationFlyout />);
-    expect(wrapper).toMatchSnapshot();
-  });
+jest.mock('../../../util/dependency_cache', () => ({
+  getToastNotifications: () => ({ addSuccess: jest.fn(), addDanger: jest.fn() }),
+}));
 
-  test('Update button is disabled with empty annotation', () => {
+const annotationUpdatesService = new AnnotationUpdatesService();
+
+const MlAnnotationUpdatesContextProvider = ({ children }: { children: React.ReactElement }) => {
+  return (
+    <MlAnnotationUpdatesContext.Provider value={{ annotationUpdatesService }}>
+      <IntlProvider>{children}</IntlProvider>
+    </MlAnnotationUpdatesContext.Provider>
+  );
+};
+
+// useObservable wraps the observable in a new component
+const ObservableComponent = (props: any) => {
+  const annotationProp = useObservable(annotationUpdatesService.update$());
+  if (annotationProp === undefined) {
+    return null;
+  }
+  return (
+    <AnnotationFlyout
+      annotation={annotationProp}
+      {...props}
+      annotationUpdatesService={annotationUpdatesService}
+    />
+  );
+};
+
+describe('AnnotationFlyout', () => {
+  test('Update button is disabled with empty annotation', async () => {
     const annotation = mockAnnotations[1] as Annotation;
-    annotation$.next(annotation);
 
-    // useObservable wraps the observable in a new component
-    const ObservableComponent = (props: any) => {
-      const annotationProp = useObservable(annotation$);
-      if (annotationProp === undefined) {
-        return null;
-      }
-      return <AnnotationFlyout annotation={annotationProp} {...props} />;
-    };
-
-    const wrapper = mountWithIntl(<ObservableComponent />);
-    const updateBtn = wrapper.find('EuiButton').first();
-    expect(updateBtn.prop('isDisabled')).toEqual(true);
+    annotationUpdatesService.setValue(annotation);
+
+    render(
+      <MlAnnotationUpdatesContextProvider>
+        <ObservableComponent />
+      </MlAnnotationUpdatesContextProvider>
+    );
+    const updateBtn = screen.getByText('Update').closest('button');
```

Comment:
would be better to use a `data-test-subj` instead

Reply:
Updated here https://github.com/elastic/kibana/pull/83216/commits/1365c52b913b0c1fd556f5642d8ba74d0746c302

Reply:
Updated here https://github.com/elastic/kibana/pull/83216/commits/1365c52b913b0c1fd556f5642d8ba74d0746c302